core/site_scons/tools.py

```diff
@@ -120,3 +121,43 @@ def get_defs_for_cmake(defs: list[str | tuple[str, str]]) -> list[str]:
         else:
             result.append(d)
     return result
+
+
+def _compress(data: bytes) -> bytes:
+    z = zlib.compressobj(level=9, wbits=-10)
+    return z.compress(data) + z.flush()
+
+
+def embed_binary(obj_program, env, section, target, file):
+    file_path = PROJECT_ROOT / Path(file)
+
+    with open(file_path, "rb") as f:
+        data = f.read()
+        compressed = _compress(data)
```

Comment:
quick tip:
```suggestion
    compressed = _compress(file_path.read_bytes())
```
(also `write_bytes()` `write_text()` `read_text()`)